File: flutter_app/lib/ui/game_over_screen.dart
import 'package:flutter/material.dart';
import '../core/palette.dart';

class GameOverScreen extends StatelessWidget {
  final int distanceMeters;
  final int moneyTaka;
  final int maxCombo;
  final String zoneName;
  final VoidCallback onPlayAgain;
  final VoidCallback onGoHome;

  const GameOverScreen({
    required this.distanceMeters,
    required this.moneyTaka,
    required this.maxCombo,
    required this.zoneName,
    required this.onPlayAgain,
    required this.onGoHome,
    super.key,
  });

  @override
  Widget build(BuildContext context) {
    return Container(
      color: Palette.voidBlack,
      padding: const EdgeInsets.all(24),
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          const Text("GAME OVER", style: TextStyle(color: Palette.coral, fontSize: 28, fontFamily: 'monospace', fontWeight: FontWeight.w900)),
          const SizedBox(height: 20),
          Container(
            width: double.infinity,
            decoration: BoxDecoration(
              color: Palette.shadow,
              borderRadius: BorderRadius.circular(8),
              border: Border.all(color: Palette.bone, width: 2),
            ),
            padding: const EdgeInsets.all(20),
            child: Column(
              children: [
                _buildStatRow("DISTANCE", "${distanceMeters}m", Palette.gold),
                _buildStatRow("MONEY", "Taka$moneyTaka", Palette.amber),
                _buildStatRow("MAX COMBO", "x$maxCombo", Palette.cyan),
                _buildStatRow("ZONE", zoneName, Palette.lime),
              ],
            ),
          ),
          const SizedBox(height: 30),
          GestureDetector(
            onTap: onPlayAgain,
            child: Container(
              width: 220,
              height: 54,
              decoration: BoxDecoration(
                color: Palette.gold,
                borderRadius: BorderRadius.circular(8),
                border: Border.all(color: Palette.ink, width: 2),
              ),
              padding: const EdgeInsets.all(8),
              child: const Center(
                child: Text("PLAY AGAIN", style: TextStyle(color: Palette.ink, fontSize: 16, fontFamily: 'monospace', fontWeight: FontWeight.bold)),
              ),
            ),
          ),
          const SizedBox(height: 14),
          _buildButton("HOME", Palette.slate, Palette.paper, onGoHome),
        ],
      ),
    );
  }

  Widget _buildStatRow(String label, String value, Color valueColor) {
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 8),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: [
          Text(label, style: const TextStyle(color: Palette.bone, fontSize: 14, fontFamily: 'monospace')),
          Text(value, style: TextStyle(color: valueColor, fontSize: 16, fontFamily: 'monospace', fontWeight: FontWeight.bold)),
        ],
      ),
    );
  }

  Widget _buildButton(String text, Color bg, Color textColor, VoidCallback onTap) {
    return GestureDetector(
      onTap: onTap,
      child: Container(
        width: double.infinity,
        height: 50,
        decoration: BoxDecoration(
          color: bg,
          borderRadius: BorderRadius.circular(6),
          border: Border.all(color: Palette.ink, width: 2),
        ),
        child: Center(
          child: Text(text, style: TextStyle(color: textColor, fontSize: 16, fontFamily: 'monospace', fontWeight: FontWeight.bold)),
        ),
      ),
    );
  }
}

File: app/src/main/java/com/example/ui/GameOverScreen.kt
package com.example.ui

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.asImageBitmap
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.core.Palette
import com.example.sprites.PixelSpriteRenderer

import androidx.compose.foundation.clickable

@Composable
fun GameOverScreen(
    distanceMeters: Int,
    moneyTaka: Int,
    maxCombo: Int,
    zoneName: String,
    onPlayAgain: () -> Unit,
    onGoHome: () -> Unit
) {
    val voidBlack = Palette.voidBlack
    val coral = Palette.coral
    val amber = Palette.amber
    val gold = Palette.gold
    val paper = Palette.paper
    val bone = Palette.bone
    val ink = Palette.ink

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(voidBlack)
            .padding(24.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        // --- GAME OVER TITLE ---
        Text(
            text = "G A M E   O V E R",
            color = coral,
            fontSize = 28.sp,
            fontWeight = FontWeight.Black,
            fontFamily = FontFamily.Monospace,
            modifier = Modifier.padding(bottom = 20.dp)
        )

        // --- STATS BOARD ---
        Box(
            modifier = Modifier
                .fillMaxWidth(0.9f)
                .background(Palette.shadow, RoundedCornerShape(8.dp))
                .border(2.dp, bone, RoundedCornerShape(8.dp))
                .padding(20.dp)
        ) {
            Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
                StatRow(label = "DISTANCE", value = "${distanceMeters} m", valueColor = gold)
                StatRow(label = "MONEY", value = "৳$moneyTaka", valueColor = amber)
                StatRow(label = "MAX COMBO", value = "x$maxCombo", valueColor = Palette.cyan)
                StatRow(label = "ZONE", value = zoneName, valueColor = Palette.lime)
            }
        }

        Spacer(modifier = Modifier.height(30.dp))

        // Pre-rendered Bangla "আবার চালাই" Play Again Button
        val playAgainBm = PixelSpriteRenderer.getBanglaTextSprite("abar_chalai")
        Image(
            bitmap = playAgainBm.asImageBitmap(),
            contentDescription = "আবার চালাই Play Again",
            modifier = Modifier
                .width(220.dp)
                .height(54.dp)
                .testTag("play_again_button")
                .background(gold, RoundedCornerShape(8.dp))
                .border(2.dp, ink, RoundedCornerShape(8.dp))
                .padding(8.dp)
                .align(Alignment.CenterHorizontally)
                .clickable { onPlayAgain() }
        )

        Spacer(modifier = Modifier.height(14.dp))

        RetroArcadeButton(
            text = "🏠  H O M E",
            bgColor = Palette.slate,
            textColor = paper,
            testTag = "home_button",
            onClick = onGoHome
        )
    }
}

@Composable
private fun StatRow(label: String, value: String, valueColor: androidx.compose.ui.graphics.Color) {
    Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically
    ) {
        Text(
            text = label,
            color = Palette.bone,
            fontSize = 14.sp,
            fontFamily = FontFamily.Monospace
        )
        Text(
            text = value,
            color = valueColor,
            fontSize = 16.sp,
            fontWeight = FontWeight.Bold,
            fontFamily = FontFamily.Monospace
        )
    }
}
